worksheet.rb

```diff
@@ -31,4 +31,147 @@
 # - the total amount of money each driver has made
 # - the average rating for each driver
 # - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+# - Which driver has the highest average rating?
+
+rides =
+    {
+        dr0001: [
+            {
+                date: "3rd Feb 2016",
+                cost: 10,
+                rider: "RD0003",
+                rating: 3
+            },
+            {
+                date: "3rd Feb 2016",
+                cost: 30,
+                rider: "RD0015",
+                rating: 4
+            },
+            {
+                date: "5th Feb 2016",
+                cost: 45,
+                rider: "RD0003",
+                rating: 2
+            }
+        ],
+        dr0002: [
+            {
+                date: "3rd Feb 2016",
+                cost: 25,
+                rider: "RD0073",
+                rating: 5
+            },
+            {
+                date: "4th Feb 2016",
+                cost: 15,
+                rider: "RD0013",
+                rating: 1
+            },
+            {
+                date: "5th Feb 2016",
+                cost: 35,
+                rider: "RD0066",
+                rating: 3
+            }
+        ],
+        dr0003: [
+            {
+                date: "4th Feb 2016",
+                cost: 5,
+                rider: "RD0066",
+                rating: 5
+            },
+            {
+                date: "5th Feb 2016",
+                cost: 50,
+                rider: "RD0003",
+                rating: 2
+            }
+        ],
+        dr0004: [
+            {
+                date: "3rd Feb 2016",
+                cost: 5,
+                rider: "RD0022",
+                rating: 5
+            },
+            {
+                date: "4th Feb 2016",
+                cost: 10,
+                rider: "RD0022",
+                rating: 4
+            },
+            {
+                date: "5th Feb 2016",
+                cost: 20,
+                rider: "RD0073",
+                rating: 5
+            }
+        ]
+    }
+
+#Iterate through the data structure to display, by printing in terminal, the following info
+# The number of rides each driver has given
+# count each array
+puts "Number of rides each driver has given:"
+rides.each do |driver,trips|
+  puts "#{driver} has given #{trips.length} rides."
+end
+
+puts "______________________________________________________________"
+puts "**************************************************************"
+
+#The total amount of money each driver has made
+# for each driver loop over the array and give me the cost and add it to sum
+# which driver made the most money
+# optional: for each driver, on which day did they make the most money?
+puts "The total amount of money each driver has made:"
+max = 0
+max_driver = ""
+#date_max_money = ""
+ rides.each do |driver,trips|
```

Comment:
Very minor, but you indented this `each` with one space. Typical Ruby code style things that things should be indented either 0 spaces, 2 spaces, 4 spaces, etc. It would probably be good to unindent this line.